keras_cv/models/segmentation/deeplab_v3_plus/deeplab_v3_plus_test.py

```diff
@@ -0,0 +1,149 @@
+# Copyright 2023 The KerasCV Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import os
+
+import pytest
+import tensorflow as tf
+from absl.testing import parameterized
+from tensorflow import keras
+
+from keras_cv.models import ResNet18V2Backbone, DeepLabV3Plus
+
+
+class DeepLabV3PlusTest(tf.test.TestCase, parameterized.TestCase):
+    @pytest.fixture(autouse=True)
+    def cleanup_global_session(self):
+        # Code before yield runs before the test
+        tf.config.set_soft_device_placement(False)
+        yield
+        # Reset soft device placement to not interfere with other unit test
+        # files
+        tf.config.set_soft_device_placement(True)
+        keras.backend.clear_session()
+
+    def test_retinanet_construction(self):
+        backbone = ResNet18V2Backbone(input_shape=[96, 96, 3])
+        model = DeepLabV3Plus(num_classes=1, backbone=backbone)
+        model.compile(
+            optimizer="adam",
+            loss=keras.losses.BinaryCrossentropy(),
+            metrics=["accuracy"],
+        )
+
+    @pytest.mark.large
+    def test_retinanet_call(self):
+        backbone = ResNet18V2Backbone(input_shape=[96, 96, 3])
+        model = DeepLabV3Plus(num_classes=1, backbone=backbone)
+        images = tf.random.uniform((2, 512, 512, 3))
+        _ = model(images)
+        _ = model.predict(images)
+
+    def test_weights_contained_in_trainable_variables(self):
+        target_size = [512, 512]
+        images = tf.ones(shape=[1] + target_size + [3])
+
+        backbone = ResNet18V2Backbone(input_shape=target_size + [3])
+        model = DeepLabV3Plus(num_classes=1, backbone=backbone)
+
+        model.compile(
+            optimizer="adam",
+            loss=keras.losses.BinaryCrossentropy(),
+            metrics=["accuracy"],
+        )
+
+        variable_names = [x.name for x in model.trainable_variables]
+        outputs = model(images)
+
+        # encoder
+        self.assertIn("conv2d_1/kernel:0", variable_names)
+        # segmentation head
+        self.assertIn("segmentation_head_conv/kernel:0", variable_names)
+        # Output shape
+        self.assertEqual(outputs.shape, tuple([1] + target_size + [1]))
+
+    @pytest.mark.large
+    def test_no_nans(self):
```

Comment:
This test is less relevant to this model than it is to RetinaNet, where this came from. It's still a reasonable assertion, but it's a lot of duplicated code from the test below. How about we just add 

```
self.assertFalse(tf.math.reduce_any(tf.math.is_nan(w2)))
```

to the assertion loop in `test_weights_change`